### PR TITLE
fix for erroroneous limit on D-M parameters

### PR DESCRIPTION
**Pull request by Rick-Methot-NOAA**
limit is now removed and the label for D-M parameters is changed.  see issue #183

### CODE REVIEW

File: SS_readcontrol_330.tpl [hunk 1]
```diff
@@ -3111,8 +3111,12 @@
     {
       echoinput<<Comp_Err_ParmCount<<"  #_parameters are needed"<<endl;
       Comp_Err_Parm_Start=N_selparm;
-      for(f=1;f<=Comp_Err_ParmCount;f++)
-        {N_selparm++; ParCount++; ParmLabel+="ln(DM_theta)_"+NumLbl(f);}
+      for (f=1;f<=Nfleet;f++) {
+      if (DM_parmlist(f)>0) {
+       N_selparm++; ParCount++; ParmLabel+="ln(DM_theta)_Len_"+NumLbl(f)+"_P"+NumLbl(Comp_Err_L2(f));}}
```
Comment (iantaylor-NOAA) [inline SS_readcontrol_330.tpl:3116]: addition of both fleet number and parameter number in parameter label will be useful for r4ss as the status-quo system relies on reading the data file to get the assignment of parameter number to fleet (discussed in https://github.com/r4ss/r4ss/issues/287).

**Comment by k-doering-NOAA** [comment]
@Rick-Methot-NOAA, looks like the 2 hake models didn't run. Maybe the fix accidentally broke DM for other situations? Just a thought since I think the hake models use DM also.

File: SS_readdata_330.tpl [hunk 2]
```diff
@@ -1746,15 +1749,15 @@
           N_warn++;  warning<<N_warn<<" minimum sample size for age comps must be > 0; minimum sample size set to 0.001 "<<endl;
           min_sample_size_A(f) = 0.001;
         }
-      if (Comp_Err_A2(f) >Nfleet)
+      if (Comp_Err_A2(f) >2*Nfleet)
       {
-        N_warn++; cout<<"fatal input error, see warning "<<endl; warning<<N_warn<<"; Age D-M index for fleet: "<<f<<" is: "<<Comp_Err_A2(f)<<" but must be an integer <=Nfleet "<<endl;
+        N_warn++; cout<<"fatal input error, see warning "<<endl; warning<<N_warn<<"; Age D-M index for fleet: "<<f<<" is: "<<Comp_Err_L2(f)<<" but must be an integer <=2*Nfleet "<<endl;
```
Comment (iantaylor-NOAA) [inline SS_readdata_330.tpl:1754]: shouldn't this stay `Comp_Err_A2` instead of changing to `Comp_Err_L2`?

**Comment by Rick-Methot-NOAA** [comment]
Working on it.  looks like I broke it for models that have age DM, but no length DM

**Review by iantaylor-NOAA — COMMENTED** [review]
Looks good overall. There may be a little work needed in r4ss to make sure the new improved parameter labels don't break the code. In the future those labels can definitely simplify the r4ss code.

There is one change that didn't make sense to me (see comment on SS_readdat_3.30.tpl).

**Pull request merged**